### ISSUE TITLE
Generics on annotated classes fail

### ISSUE THREAD

**Issue by peckb1** [peckb1/autojackson#20]
If an annotated class has generics, such as:
```
@AutoJackson(type = @AutoJackson.Type(FraggleName.class))
public interface Fraggle<M extends Muppeteer> {

    M getMuppeteer();

    enum FraggleName {
        GOBO(Gobo.class);

        private final Class<? extends Fraggle> fraggleClass;

        FraggleName(Class<? extends Fraggle> fraggleClass) {
            this.fraggleClass = fraggleClass;
        }

        @AutoJacksonTypeClass
        public Class<? extends Fraggle> getFraggleClass() {
            return fraggleClass;
        }
    }
}

@AutoJackson
public interface Gobo extends Fraggle<JerryNelson> { }
```

When the `Gobo_AutoJacksonImpl` is created, the type for `muppeteer` is `M` and not `JerryNelson`

**Comment by peckb1** [comment]
Addressed on 1.1.0-SNAPSHOT

**Comment by peckb1** [comment]
Fixed in release 1.1.0
